backend/app/api/src/helpers/AdminPermissionChecker.ts

```diff
@@ -187,26 +187,130 @@ export class AdminPermissionChecker {
+                }
+                organizationPermissions = await this.getOrganizationPermissions(organization);
+            } catch(error) {
+                console.error(error);
+                throw new SimpleError({
+                    code: 'not_found',
+                    message: 'Event not found',
+                    human: 'De activiteit werd niet gevonden',
+                })
+            }
+
+            if(organizationPermissions === null) {
+                throw new SimpleError({
+                    code: 'permission_denied',
+                    message: 'Je hebt geen toegangsrechten om een activiteit te beheren voor deze organisatie.',
+                    statusCode: 400
+                })
+            }
 
-        if (permissionLevel !== PermissionLevel.Read) {
-            if (event.organizationId) {
-                // Need full access for now
-                if (!await this.hasFullAccess(event.organizationId)) {
-                    return false
+            if(eventForGroups === null) {
+                if(!organizationPermissions.hasResourceAccessRight(PermissionsResourceType.Groups, '', accessRight)) {
+                    throw new SimpleError({
+                        code: 'permission_denied',
+                        message: 'Je hebt geen toegangsrechten om een activiteit te beheren voor deze organisatie.',
+                        statusCode: 400
+                    })
                 }
             } else {
-                if (!this.hasPlatformFullAccess()) {
-                    return false
+                for(const group of eventForGroups) {
+                    if(!organizationPermissions.hasResourceAccessRight(PermissionsResourceType.Groups, group.id, accessRight)) {
+                        throw new SimpleError({
+                            code: 'permission_denied',
+                            message: 'Je hebt geen toegangsrechten om een activiteit te beheren voor deze groep(en).',
+                            statusCode: 400
+                        })
+                    }
+                }
+            }
+
+            if(eventForTags !== null) {
+                // not supported currently
+                throw new SimpleError({
+                    code: 'invalid_field',
+                    message: 'Een activiteit voor een organisatie kan geen tags bevatten.',
+                    statusCode: 400
+                })
+            }
+
+            if(eventForDefaultAgeGroupIds !== null) {
+                // not supported currently
+                throw new SimpleError({
+                    code: 'invalid_field',
+                    message: 'Een activiteit voor een organisatie kan niet beperkt worden tot specifieke standaard leeftijdsgroepen.',
+                    statusCode: 400
+                })
+            }
+
+            return organization;
+        } else if(eventForGroups !== null) {
```

Comment:
```suggestion
        }
        
if(eventForGroups !== null) {
```

Door de else te gebruiken maak je het minder duidelijk dat er een 'return' is in de eerste 'if'